File: modules/chat-core/src/main/java/com/shterneregen/securelan/chat/discovery/PeerDiscoveryService.java
package com.shterneregen.securelan.chat.discovery;

import java.util.List;

public interface PeerDiscoveryService extends AutoCloseable {
    void start(PeerDiscoveryConfig config, PeerDiscoveryListener listener);

    void stop();

    void setAnnounceEnabled(boolean announceEnabled);

    boolean isRunning();

    List<DiscoveredPeer> snapshot();

    @Override
    default void close() {
        stop();
    }
}

File: modules/chat-core/src/main/kotlin/com/shterneregen/securelan/chat/discovery/PeerDiscoveryService.kt
package com.shterneregen.securelan.chat.discovery

interface PeerDiscoveryService : AutoCloseable {
    fun start(config: PeerDiscoveryConfig, listener: PeerDiscoveryListener)

    fun stop()

    fun setAnnounceEnabled(announceEnabled: Boolean)

    fun isRunning(): Boolean


    fun snapshot(): List<DiscoveredPeer>

    override fun close() {
        stop()
    }
}
